proposals/NNNN-set-language-separately.md

```diff
@@ -0,0 +1,55 @@
+# Set language separately
+
+* Proposal: [SDL-NNNN](NNNN-set-language-separately.md)
+* Author: [zhouxin627](https://github.com/zhouxin627)
+* Status: **Awaiting review**
+* Impacted Platforms: [iOS / Java Suite]
+
+## Introduction
+This proposal provides support for updating to two separate languages.
+
+## Motivation
+According to RPC Spec documents, it seems like when the HU's `hmiDisplayLanguage` or `VR/TTS` language is different from the app's desired language, the app will send `ChangeRegistration` to change the registered language which is supported.
+But at the SDL Proxy sourcecode's side, only the difference of `VR/TTS` language will trigger `ChangeRegistration`.
+The Proxy does not currently support updating to two separate languages.
+In other words, the developer can only be told that the language updated, but not which language type.
+The `hmiDisplayLanguage` and `VR/TTS` language could be set separately in some use cases for OEM.
+
+## Proposed solution
+The proposed solution is to add a new method to pass the HMI display language separately, and the old one should be deprecated.
+
+### iOS Library
+A new SDLManagerDelegate method would need to be added, and the old one should be deprecated:
+
+```objc
+	 * Called when the lifecycle manager detected a language mismatch. In case of a language mismatch the manager should change the apps registration by updating the lifecycle configuration to the specified language. If the app can support the specified language it should return an Object of SDLLifecycleConfigurationUpdate, otherwise it should return nil to indicate that the language is not supported.

```

Comment:
```suggestion
	 * Called when the lifecycle manager detected a language mismatch. In case of a language mismatch the manager should change the app's registration by updating the lifecycle configuration to the specified language. If the app can support the specified language it should return an Object of SDLLifecycleConfigurationUpdate, otherwise it should return nil to indicate that the language is not supported.

```